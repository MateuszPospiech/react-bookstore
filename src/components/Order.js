import React from 'react';

class Order extends React.Component {

    render(){

        return (
            <div className="order col-4 col-xs-12">
                Zam√≥wienie
            </div>
        )
    }
}

export default Order;