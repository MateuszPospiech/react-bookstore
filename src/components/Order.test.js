import React from 'react';
import ReactDOM from 'react-dom';
import Order from './Order';

import {configure, shallow} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({adapter: new Adapter()});

describe('App tests', () => {

    it('App renders without a problem', () => {
        const div = document.createElement('div');
        ReactDOM.render(<Order />, div);
        ReactDOM.unmountComponentAtNode(div);
    })

    it ('Hello world renders', () =>{
        const wrapper = shallow(<Order/>);
        // console.log(wrapper.debug());
        expect(wrapper.find('div').text()).toBe('ZamÃ³wienie')
    })

    it ('Snapshot watches', () =>{
        const wrapper = shallow(<Order/>);
        expect(wrapper).toMatchSnapshot();
    })
})