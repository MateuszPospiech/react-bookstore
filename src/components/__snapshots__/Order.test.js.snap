// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App tests Snapshot watches 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Order />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": "Zam贸wienie",
      "className": "order col-4 col-xs-12",
    },
    "ref": null,
    "rendered": "Zam贸wienie",
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "Zam贸wienie",
        "className": "order col-4 col-xs-12",
      },
      "ref": null,
      "rendered": "Zam贸wienie",
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
